### PR TITLE
Use ui core component combo box in stdcm

### LINKED ISSUE #8706: Integrate InputWithSuggestions

### PR DESCRIPTION
**Pull request by achrafmohye**
Closes #8706

### CODE REVIEW

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx [hunk 1]
```diff
@@ -15,7 +15,6 @@ import type { StdcmConfSliceActions } from 'reducers/osrdconf/stdcmConf';
 import { useAppDispatch } from 'store';
 
 import StdcmCard from './StdcmCard';
-import StdcmSuggestions from './StdcmSuggestions';
 import type { StdcmConfigCardProps } from '../types';
 
 type StdcmSuggestionsConsistOption = SelectOptionObject &
```
Comment (Uriel-Sautron) [inline front/src/applications/stdcmV2/components/StdcmConsist.tsx]: This type is not used

File: front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx [hunk 2]
```diff
@@ -131,9 +108,14 @@ const StdcmOperationalPoint = ({
...
-    setSearchTerm(suggestionName);
-    updateSelectedPoint(sortedSearchResults, uic);
+  const onSelectSuggestion = ({
+    value: suggestionName,
+    uic,
+  }: { value?: string; uic?: number } = {}) => {
+    setSearchTerm(suggestionName || '');
+    if (uic) {
+      updateSelectedPoint(sortedSearchResults, uic);
+    }
```
Comment (clarani) [inline front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx]: Only a suggestion, but I find it clearer to manage the case where the selectedSuggestion is undefined like this (the types used are easier to understand too)

```suggestion
   const onSelectSuggestion = (selectedSuggestion?: Option) => {
    if (!selectedSuggestion) {
      setSearchTerm('');
      return;
    }
    const { value: suggestionName, uic } = selectedSuggestion;
    setSearchTerm(suggestionName);
    updateSelectedPoint(sortedSearchResults, uic);
```

**Review by clarani — APPROVED** [review]
LGTM, tested :white_check_mark:

**Review by kmer2016 — COMMENTED** [review]
Some strange behaviours/bugs:
- When you clear the selected RS using the "delete" or "backspace" key (without using the clear button), then lose focus and refocus, the suggestion list only shows the last deleted option instead of all suggestions.
- We can't find Operational point by trigram anymore

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx (hunk 1 above)
Comment (achrafmohye) [reply front/src/applications/stdcmV2/components/StdcmConsist.tsx]: Done

File: front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx (hunk 2 above)
Comment (achrafmohye) [reply front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx]: Done :)

**Review by kmer2016 — APPROVED** [review]

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx [hunk 3]
```diff
@@ -75,18 +68,13 @@ const StdcmConsist = ({ setCurrentSimulationInputs, disabled = false }: StdcmCon
...
-    } else {
-      dispatch(updateRollingStockID(undefined));
+    if (e.target.value.trim().length === 0) {
+      updateRollingStockID(undefined);
     }
   };
 
-  const onSelectSuggestion = (option: StdcmSuggestionsConsistOption) => {
-    dispatch(updateRollingStockID(option.value.id));
+  const onSelectSuggestion = (option: LightRollingStockWithLiveries | undefined) => {
```
Comment (SharglutDev) [inline front/src/applications/stdcmV2/components/StdcmConsist.tsx]:
```suggestion
  const onSelectSuggestion = (option?: LightRollingStockWithLiveries) => {
```

File: front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
-import StdcmSuggestions from './StdcmSuggestions';
-
 type StdcmOperationalPointProps = {
   updatePoint: (pathStep: PathStep | null) => void;
   point: PathStep | null;
   opPointId: string;
   disabled?: boolean;
 };
 
+export type Option = { label: string; value: string; uic: number };
```
Comment (SharglutDev) [inline front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx]: Do we use it outside of this file ?

Comment (achrafmohye) [reply front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx]: It was used before in a helper not anymore, nice remark, resolved :)

File: front/src/applications/stdcmV2/components/StdcmConsist.tsx (hunk 3 above)
Comment (achrafmohye) [reply front/src/applications/stdcmV2/components/StdcmConsist.tsx]: Done :)

**Review by SharglutDev — APPROVED** [review]
Lgtm (not tested)

**Review by Uriel-Sautron — APPROVED** [review]

**Pull request merged**